components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/dao/impl/WebhookManagementDAOFacade.java

```diff
@@ -0,0 +1,331 @@
+import org.wso2.carbon.database.utils.jdbc.exceptions.TransactionException;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.webhook.management.api.constant.ErrorMessage;
+import org.wso2.carbon.identity.webhook.management.api.exception.WebhookMgtException;
+import org.wso2.carbon.identity.webhook.management.api.model.Webhook;
+import org.wso2.carbon.identity.webhook.management.api.model.WebhookStatus;
+import org.wso2.carbon.identity.webhook.management.api.service.EventSubscriberService;
+import org.wso2.carbon.identity.webhook.management.internal.component.WebhookManagementComponentServiceHolder;
+import org.wso2.carbon.identity.webhook.management.internal.dao.WebhookManagementDAO;
+import org.wso2.carbon.identity.webhook.management.internal.util.WebhookManagementExceptionHandler;
+
+import java.util.List;
+
+/**
+ * Facade implementation for WebhookManagementDAO.
+ * This class coordinates between webhook subscriber services and webhook management DAO.
+ * It ensures that webhooks are first subscribed with external systems before being persisted.
+ */
+public class WebhookManagementDAOFacade implements WebhookManagementDAO {
+
+    private static final Log LOG = LogFactory.getLog(WebhookManagementDAOFacade.class);
+
+    private final WebhookManagementDAO webhookManagementDAO;
+
+    /**
+     * Constructor with WebhookManagementDAO.
+     *
+     * @param webhookManagementDAO WebhookManagementDAO instance.
+     */
+    public WebhookManagementDAOFacade(WebhookManagementDAO webhookManagementDAO) {
+
+        this.webhookManagementDAO = webhookManagementDAO;
+    }
+
+    /**
+     * Create a new webhook subscription in the database.
+     * First checks if a webhook with same endpoint exists, then subscribes with the webhook subscriber service,
+     * and finally persists it using the DAO.
+     *
+     * @param webhook  Webhook subscription to be created.
+     * @param tenantId Tenant ID.
+     * @throws WebhookMgtException If an error occurs while creating the webhook subscription.
+     */
+    @Override
+    public void createWebhook(Webhook webhook, int tenantId) throws WebhookMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
+        EventSubscriberService subscriberService =
+                WebhookManagementComponentServiceHolder.getInstance().getEventSubscriberService();
+
+        try {
+            jdbcTemplate.withTransaction(template -> {
+                boolean subscribed = subscriberService.subscribe(webhook, tenantDomain);
+                if (!subscribed) {
+                    LOG.warn("Failed to subscribe webhook: " + webhook.getId());
+                    webhook.setStatus("INACTIVE");
+                }
```

Comment:
Well, if we handle failure subscriptions in this manner and go and create the entry in db, it makes things hard to manage.

If some thing fails and topics have not been created or subscription is not created, with this mode, there should be an entry point to retrigger same without going over webhook creation workflow. We don't have something like that

IMO, if the subscription fails this entry should not be created and it should treated as a server error. so the user can start creating again